### PR TITLE
Update: recursion addendum section

### PR DESCRIPTION
**Pull request by jordalgo**
Make a note about the calling of `read_filesystem` instead of `read_directory` to get the whole solution working.

### CODE REVIEW

File: site/learn/tutorials/if_statements_loops_and_recursion.md [hunk 1]
```diff
@@ -784,7 +784,14 @@ and read_directory path =
   list
 ```
 Notice `End_of_file -> entries` which means "when we get the exception,
-give up and return the result so far".
+give up and return the result so far". Also, note that the function name
```
Comment (Chris00) [inline site/learn/tutorials/if_statements_loops_and_recursion.md]: What function name?

File: site/learn/tutorials/if_statements_loops_and_recursion.md [hunk 2]
```diff
@@ -784,7 +784,14 @@ and read_directory path =
...
+has changed and you're invoking of it will have to change too:
```
Comment (Chris00) [inline site/learn/tutorials/if_statements_loops_and_recursion.md]: > you're invoking of it

Is this proper English?

File: site/learn/tutorials/if_statements_loops_and_recursion.md [hunk 3]
```diff
@@ -784,7 +784,14 @@ and read_directory path =
...
+
+```ocaml
+let path = Sys.argv.(1) in
+let fs = read_filesystem path in
+print_endline (string_of_filesystem fs)
+```
```
Comment (Chris00) [inline site/learn/tutorials/if_statements_loops_and_recursion.md:795]: I do not think the text you added is particularly clear.  May you please improve it?

**Review by Chris00 — CHANGES_REQUESTED** [review]

**Comment by jordalgo** [comment]
@Chris00 Updated the text I added. Hopefully it's more clear. Not sure why my diff is showing the addition of spaces. Let me know if you need me to delete those.

**Comment by Chris00** [comment]
> Let me know if you need me to delete those.

Please do.

**Comment by jordalgo** [comment]
Done.

**Comment by samoht** [comment]
@Chris00 are you happy with the new changes?

**Comment by avsm** [comment]
Thanks for the iteration @jordalgo! I think the newer version is clearer; we can refine it with a followup PR if necessary.

**Pull request merged**